script.js

```diff
@@ -1,4 +1,365 @@
+    console.log("else if statement", computerPlays, userGuess);
+  }
+  return myOutputValue;
+};
+
+//=========================== MAIN FUNC ============================
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  //Prevent caps errors
+  input = input.toLowerCase();
+  var myOutputValue = ``;
+
+  var computerPlays = generateComputerResult();
+  console.log(
+    `generateComputerResult function, computer plays: ${computerPlays}`
+  );
+
+  var standardMessage = generateStdMsg(userName, computerPlays, input);
+  console.log(standardMessage);
+
+  //determine the game mode
+  if (currentMode == `pending inputs`) {
+    userName = input;
+    console.log(`user name: ${userName}`);
+    // now that we have the name, switch the mode
+    currentMode = "normal game mode";
+    myOutputValue = `Hello ${userName}! <br> Enter 'scissors', 'paper' or 'stone' to start the game! <br><br> Alternatively you can input any of the following to change the game mode: <br> 
+    'reversed', 'korean' or 'com v com'.`;
+    console.log(`current normal mode: ${currentMode}`);
+  } else if (currentMode != `pending inputs` && input == "normal") {
+    currentMode = "normal game mode";
+    myOutputValue = `Let's play normal SPS!`;
+
+    //allow selection of game modes
+  } else if (currentMode != `pending inputs` && input == "korean") {
+    currentMode = `mukJjiPpa`;
+    myOutputValue = `Let's play Muk-Jji-Ppa!`;
+  } else if (currentMode != `pending inputs` && input == `reversed`) {
+    currentMode = `reversed`;
+    myOutputValue = `Let's play reversed SPS!`;
+  } else if (currentMode != `pending inputs` && input == `com v com`) {
+    currentMode = `com v com`;
+    myOutputValue = `Let's play Computer VS Computer!`;
+
+    //start of actual game
+  } else if (currentMode == "normal game mode") {
+    console.log(`current mode: ${currentMode}`);
+    myOutputValue = `${
+      playNormalGame(input, computerPlays, standardMessage)}, <br>${overallResult}`;
+    //introduce reversed game mode
+  } else if (currentMode == "reversed") {
+    console.log(`current mode: ${currentMode}`);
+    myOutputValue = `${
+       playReversedGame(input, computerPlays, standardMessage)}, <br>${overallResult}`;
+    // introduce muk-jji-ppa game mode
+  } else if (currentMode == `mukJjiPpa`) {
+    console.log(`current mode: ${currentMode}`);
+    myOutputValue = `${
+      playKoreanGame(input, computerPlays, standardMessage)} <br>${overallResult}`;
+  } else if (currentMode == `com v com`) {
+    console.log(`current mode: ${currentMode}`);
+    myOutputValue = `${
+      playCOMvCOMgame(input, computerPlays, standardMessage)}<br> ${overallResult}`;
+  }
+
+  var overallResult = trackOverallResult(userWins, compWins);
+  console.log(`User and comp wins : ${(userWins, compWins,overallResult)}`);
+
   return myOutputValue;
```

Comment:
I would recommend to put this higher along with your other helper functions above the game mode section if possible! It would be best to group all similar functions that do similar things together.